File: python/day01/part1.py
#!/usr/bin/env python3
import time


def parse(input_path):
    res = []
    curr = 0
    with open(input_path, 'r') as f:
        for line in f.readlines():
            if len(line.strip()) == 0:
                res.append(curr)
                curr = 0
            else:
                curr += int(line)
    return res


def solve_part1(arr):
    return max(arr)


if __name__ == "__main__":
    start = time.time()
    parsed = parse("data.txt")
    result = solve_part1(parsed)
    print(f'Part 1 : {result}  ({round(1000 * (time.time() - start))} ms)')

File: c++/day01/part1.cpp
#include "part1.h"

using namespace std;


namespace Day01 {

    vector<int> parse(const string& filename) {
        vector<int> parsed;
        int curr = 0;
        for (const string& line : getFileLines(filename)) {
            if (trim(line).size()) {
                curr += stoi(line);
            } else {
                parsed.push_back(curr);
                curr = 0;
            }
        }
        return parsed;
    }

    namespace Part1 {
        int solve(const vector<int>& parsed) {
            return *max_element(parsed.begin(), parsed.end());
        }
    }
}
